terraform/_gen/plural_resource.go.tpl

```diff
@@ -191,7 +191,7 @@ func (r resourceTeleport{{.Name}}) Read(ctx context.Context, req tfsdk.ReadResou
 	}
 
 	var id types.String
-	diags = req.State.GetAttribute(ctx, path.Root("metadata").AtName("name"), &id)
+	diags = req.State.GetAttribute(ctx, {{if .MetadataNameAsID -}}path.Root("id"){{else -}}path.Root("metadata").AtName("name"){{end}}, &id)
```

Comment:
Am I reading this right? You're reading into the `id` variable, and `if .MetadataNameAsID` you get `root.id`, else you get `root.metadata.name`? It seems backward

Reply:
This field is no longer required as embedded fields are supported now (see https://github.com/gravitational/protoc-gen-terraform/pull/29). I have updated this PR.